public/pages/Findings/containers/Findings/Findings.tsx

```diff
@@ -157,44 +157,57 @@ class Findings extends Component<FindingsProps, FindingsState> {
     this.setState({ loading: true });
     const { findingsService, detectorService, notifications } = this.props;
     try {
-      const detectorsRes = await detectorService.getDetectors();
-      if (detectorsRes.ok) {
-        const detectors = detectorsRes.response.hits.hits;
+      // Get all the Findings logic
+      const findingRes = await findingsService.getFindings();
+
+      if (findingRes.ok) {
         const ruleIds = new Set<string>();
         let findings: FindingItemType[] = [];
 
-        const detectorId = this.props.match.params['detectorId'];
-        for (let detector of detectors) {
-          if (!detectorId || detector._id === detectorId) {
-            const findingRes = await findingsService.getFindings({ detectorId: detector._id });
-
-            if (findingRes.ok) {
-              const detectorFindings: FindingItemType[] = findingRes.response.findings.map(
-                (finding) => {
-                  finding.queries.forEach((rule) => ruleIds.add(rule.id));
-                  return {
-                    ...finding,
-                    detectorName: detector._source.name,
-                    logType: detector._source.detector_type,
-                    detector: detector,
-                  };
-                }
-              );
-              findings = findings.concat(detectorFindings);
-            } else {
-              errorNotificationToast(notifications, 'retrieve', 'findings', findingRes.error);
+        // Assuming getDetectors returns an array of detectors
+        const detectorsRes = await detectorService.getDetectors();
+        if (detectorsRes.ok) {
+          const detectors = detectorsRes.response.hits.hits;
+          const detectorInfoMap = new Map();
+
+          detectors.forEach((detector) => {
+            detectorInfoMap.set(detector._id, {
+              logType: detector._source.detector_type,
+              name: detector._source.name,
+              detector: detector,
+            });
+          });
+          const detectorFindingsPromises = findingRes.response.findings.map(async (finding) => {
```

Comment:
Does this need to be `async`? On line 161, we `await` the response of the `getFindings` API call, so I don't believe we'll be dealing with promises here (correct me if I'm missing something though).

Reply:
yes right, it does not necessarily need to be async because we're working with the result of an already awaited API call (findingsService.getFindings()).